File: Sprint1/I/C++/main.cpp
// https://contest.yandex.ru/contest/22449/problems/I/
#include <iostream>

bool is_four_degree(int n) {
    if (n == 1) {
        return true;
    }
    if (n % 4 != 0) {
        return false;
    } else {
        return is_four_degree(n >> 2);
    }
}

int main() {
    int n;
    std::cin >> n;
    if (is_four_degree(n)) {
        std::cout << "True" << std::endl;
    } else {
        std::cout << "False" << std::endl;
    }

    return 0;
}

File: Sprint1/I/Rust/src/main.rs
// https://contest.yandex.ru/contest/22449/problems/I/

use std::io;

fn is_four_degree(n: i32) -> bool {
    match n {
        1 => true,
        _ if n % 4 != 0 => false,
        _ => is_four_degree(n >> 2),
    }
}

fn main() -> io::Result<()> {
    let mut input = String::new();
    io::stdin().read_line(&mut input)?;

    let n = input.trim().parse::<i32>().unwrap();
    if is_four_degree(n) {
        println!("True");
    } else {
        println!("False");
    }

    Ok(())
}
